modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/QuickTreeBreak.pm

```diff
@@ -369,10 +367,10 @@ sub generate_subtrees {
                 $max_subtree = $child;
             }
         }
-        # Broke down to half, happy with it
-        print STDERR "QuickTreeBreak iterate -- $max_num_leaves (goal: $half_count)\n"; # if ($self->debug);
+        # Broke down to half or to the specified treebreak_gene_count, happy with it
+        print STDERR "QuickTreeBreak iterate -- $max_num_leaves (goal: $half_count)\n";
         die "Tree is empty - cannot break" if $max_num_leaves == 0;
-        if ($max_num_leaves <= $half_count) {
+        if (($max_num_leaves <= $half_count) or ($max_num_leaves <= ($self->param('treebreak_gene_count')))) {
```

Comment:
It may require more code changes, but what about renaming `$half_count` to `$count_threshold`, and define it as `$count_threshold = min(int(scalar(@{$members})/2), $self->param('treebreak_gene_count'));`? I feel it would make the code slightly more readable from the beginning (including the previous print, that prints `$half_count` as the goal when it will not always be the case anymore), but I will leave it up to you.

Reply:
I agree with you, but I am scared to change it now for fear of breaking anything and have no time for further testing! We leave this short exchange as historic acknowledgement for the next person who needs to dig into the code :nerd_face: